src/plot.cpp

```diff
@@ -1027,23 +1029,45 @@ RGBColor random_color(void)
     int(prn(&model::plotter_seed) * 255), int(prn(&model::plotter_seed) * 255)};
 }
 
-ProjectionPlot::ProjectionPlot(pugi::xml_node node) : PlottableInterface(node)
+RayTracePlot::RayTracePlot(pugi::xml_node node) : PlottableInterface(node)
 {
-  set_output_path(node);
   set_look_at(node);
   set_camera_position(node);
   set_field_of_view(node);
   set_pixels(node);
-  set_opacities(node);
   set_orthographic_width(node);
-  set_wireframe_thickness(node);
-  set_wireframe_ids(node);
-  set_wireframe_color(node);
+  set_output_path(node);
 
   if (check_for_node(node, "orthographic_width") &&
       check_for_node(node, "field_of_view"))
     fatal_error("orthographic_width and field_of_view are mutually exclusive "
                 "parameters.");
+
+  // Get centerline vector for camera-to-model. We create vectors around this
+  // that form a pixel array, and then trace rays along that.
+  auto up = up_ / up_.norm();
+  Direction looking_direction = look_at_ - camera_position_;
+  looking_direction /= looking_direction.norm();
+  if (std::abs(std::abs(looking_direction.dot(up)) - 1.0) < 1e-9)
+    fatal_error("Up vector cannot align with vector between camera position "
+                "and look_at!");
+  Direction cam_yaxis = looking_direction.cross(up);
+  cam_yaxis /= cam_yaxis.norm();
+  Direction cam_zaxis = cam_yaxis.cross(looking_direction);
+  cam_zaxis /= cam_zaxis.norm();
+
+  // Cache the camera-to-model matrix
+  camera_to_model_ = {looking_direction.x, cam_yaxis.x, cam_zaxis.x,
```

Comment:
If this is a fixed size, can we use a `std::array<double, 9>` for this attribute instead?

Reply:
This and the ray generation operations below are reinforcing my feeling that we might want to have a dedicated `Camera` struct.

Reply:
It's because `Position::rotate` takes an `std::vector` of length 9 to be interpreted as a 3x3 matrix. Not a very nice setup, but it's what we have. I can change `rotate` to be a template on any iterable, up to you.

Reply:
Probably not necessary at the moment. Thanks for being open to it!